src/check_datapackage/internals.py

```diff
@@ -134,3 +241,43 @@ def _get_full_json_path_from_error(error: ValidationError) -> str:
         if match:
             return f"{error.json_path}.{match.group(1)}"
     return error.json_path
+
+
+def _validation_error_to_schema_errors(error: ValidationError) -> Iterable[SchemaError]:
+    current = [_create_schema_error(error)]
+    if not error.context:
+        return current
+    return chain(
+        current,
+        flat_map(_validation_error_to_schema_errors, error.context),
+    )
+
+
+def _create_schema_errors(errors: list[ValidationError]) -> list[SchemaError]:
+    return list(flat_map(_validation_error_to_schema_errors, errors))
+
+
+def _create_schema_error(error: ValidationError) -> SchemaError:
+    return SchemaError(
+        message=error.message,
+        validator=str(error.validator),
+        json_path=_get_full_json_path_from_error(error),
+        schema_path="/".join(_map(error.absolute_schema_path, str)),
+        parent=_create_schema_error(error.parent) if error.parent else None,  # type: ignore[arg-type]
+    )
+
+
+def _path_or_data_required_error(error: SchemaError) -> bool:
+    return error.json_path.endswith(("path", "data")) and error.validator == "required"
+
+
+def _is_path_type_error(error: SchemaError) -> bool:
+    return error.validator == "type" and error.json_path.endswith("path")
+
+
+def _create_issue(error: SchemaError) -> Issue:
+    return Issue(
+        message=error.message,
+        location=error.json_path,
```

Comment:
```suggestion
        json_path=error.json_path,
```

We've updated this to `json_path`. You'll have to update from main to get those changes.